tests/Parallel_FilenameTest.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2016 Alexander Grund
+ *
+ * This file is part of libSplash.
+ *
+ * libSplash is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libSplash is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libSplash.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ #include <cppunit/extensions/HelperMacros.h>
+
+#include "splash/splash.h"
+#include <mpi.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+using namespace splash;
+
+class FilenameTest : public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(FilenameTest);
+
+    CPPUNIT_TEST(testBaseName);
+    CPPUNIT_TEST(testFullName);
+
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+
+    FilenameTest();
+    virtual ~FilenameTest();
+
+private:
+    /**
+     * Tests that file naming is correct when passing only base name
+     */
+    void testBaseName();
+
+    /**
+     * Tests that file naming is correct when passing full name
+     */
+    void testFullName();
+
+    void runTest(const char* filename, const char* fullFilename);
+
+    bool fileExists(const std::string& filename);
+
+    IParallelDataCollector *dataCollector;
+    ColTypeInt ctInt;
+    int mpiRank;
+    static int initCt;
+};
+
+int FilenameTest::initCt = 0;
+
+CPPUNIT_TEST_SUITE_REGISTRATION(FilenameTest);
+
+#define HDF5_FILE "h5/testParallelFilename"
+#define HDF5_BASE_FILE HDF5_FILE "_2.h5"
+#define HDF5_FULL_FILE HDF5_FILE ".h5"
+
+#define MPI_SIZE_X 1
```

Comment:
It might be possible. But this should only test the usage of the `ParallelDataCollector` for a single rank (MPI_COM_SELF) as this was how it was used before when the `SerialDataCollector` was not able to handle this. However I would not try this as there are more flaws in the implementation of libSplash. E.g. `handles.open(Dimensions(1, 1, 1), filename, fileAccProperties, H5F_ACC_RDONLY);` Note the first argument (`mpiSize`) is constant. I don't want to fix this here and hence don't want tests for that.